api/client/connection.go

```diff
@@ -340,24 +357,34 @@ func (c *Connection) IndexPost(ctx context.Context, id ksuid.KSUID, post api.Ind
 	return err
 }
 
-func (c *Connection) Load(ctx context.Context, poolID, branchID ksuid.KSUID, r io.Reader, commit api.CommitRequest) (*Response, error) {
-	encoded, err := json.Marshal(commit)
-	if err != nil {
+func (c *Connection) Load(ctx context.Context, poolID ksuid.KSUID, branchName string, r io.Reader, message api.CommitMessage) (*Response, error) {
+	req := c.Request(ctx).
+		SetBody(r)
+	if err := encodeCommitMessage(req, message); err != nil {
 		return nil, err
 	}
-	req := c.Request(ctx).
-		SetBody(r).
-		SetHeader("Zed-Commit", string(encoded))
 	req.Method = http.MethodPost
-	req.URL = path.Join("/pool", poolID.String(), branchID.String())
+	req.URL = path.Join("/pool", poolID.String(), "branch", url.QueryEscape(branchName))
 	return c.stream(req)
 }
 
-func (c *Connection) Delete(ctx context.Context, poolID, branchID ksuid.KSUID, ids []ksuid.KSUID) (*Response, error) {
+func encodeCommitMessage(req *resty.Request, message api.CommitMessage) error {
+	encoded, err := json.Marshal(message)
+	if err != nil {
+		return err
+	}
+	req.SetHeader("Zed-Commit", string(encoded))
+	return nil
+}
+
+func (c *Connection) Delete(ctx context.Context, poolID ksuid.KSUID, branchName string, ids []ksuid.KSUID, message api.CommitMessage) (*Response, error) {
 	req := c.Request(ctx).
 		SetBody(ids)
+	if err := encodeCommitMessage(req, message); err != nil {
+		return nil, err
+	}
 	req.Method = http.MethodPost
-	req.URL = path.Join("/pool", poolID.String(), branchID.String(), "delete")
+	req.URL = path.Join("/pool", poolID.String(), "branch", branchName, "delete")
```

Comment:
`branchName` should be path-encoded.

To avoid omissions like this, maybe say
```go
	req.URL = urlPath("pool", poolID.String(), "branch", branchName)
```
with
```go
func urlPath(elem ...string) string {
	var s string
	for _, e := range elem {
		s += "/" + url.PathEncode(e)
	}
	return path.Clean(s)
}
```

Reply:
Or use url.PathEscape